src/offers/lnd_requests.rs

```diff
@@ -247,15 +248,64 @@ pub(super) async fn create_invoice_info_from_request(
         payment_paths,
     })
 }
-/// create_reply_path creates a blinded path to provide to the offer node when requesting an
-/// invoice so they know where to send the invoice back to. We try to find a peer that we're
-/// connected to with the necessary requirements to form a blinded path. The peer needs two
+/// create_reply_path_for_offer_creation creates a blinded path to provide to the offer.
+/// We try to find at least one active public channel and at most three active public channels.
+///
+/// Otherwise we create a blinded path directly to ourselves.
+pub async fn create_reply_path_for_offer_creation(
+    mut connector: impl PeerConnector + std::marker::Send + 'static,
+    node_id: PublicKey,
+    message_context: MessageContext,
+    messenger_utils: &MessengerUtilities,
+) -> Result<BlindedMessagePath, OfferError> {
+    // Find introduction channels for our blinded path.
+    let current_channels = connector.list_active_public_channels().await.map_err(|e| {
+        error!("Could not lookup current channels: {e}.");
+        OfferError::ListChannelsFailure(e)
+    })?;
+
+    let mut intro_channels = vec![];
+    for channel in current_channels.channels.iter().take(3) {
```

Comment:
I think we are missing two important things:
  1.  does peer has onion support? So they can actually relay to us the onion messages.
  2. does peer has more public channels beside us so we know they are somehow connected?
  
  After filtering our peers giving those cases,  we should take top 3 best peers. If not found we should do the one hop blinded path. In the last case I'd add a debug log, because probably is not intended.